content/en/blog/_posts/2023-04-18-node-expand-secret-moves-beta

```diff
@@ -0,0 +1,128 @@
+
+How do I use CSINodeExpandSecret feature?
+
+Assuming all the required components (including CSI driver) are already deployed and running on your cluster, you can try NodeExpansion operation on a CSI volume if the CSI driver support the same. 
+
+The credentials for CSI NodeExpand Secret operation can be provided as Kubernetes Secret Object and specifying via StorageClass that use that Secret.
+
+Here is an example:
+
+manifest for a Secret that holds credentials:
+
+
+apiVersion: v1
+kind: Secret
+metadata:
+ name: test-secret
+ namespace: default
+data:
+stringData:
+ username: admin
+ password: t0p-Secret
+
+Here's an example manifest for a StorageClass that refers to those credentials:
+
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+ name: csi-blockstorage-sc
+parameters:
+ csi.storage.k8s.io/node-expand-secret-name: test-secret  # the name of the Secret
+ csi.storage.k8s.io/node-expand-secret-namespace: default # the namespace that the Secret is in
+provisioner: blockstorage.cloudprovider.example
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+allowVolumeExpansion: true
+
+Example output:
+
+If the PersistentVolumeClaim (PVC) was created successfully, you can see that configuration within the `spec.csi` field of the PersistentVolume (look for `spec.csi.nodeExpandSecretRef`). Check that it worked by running kubectl get persistentvolume <pv_name> -o yaml. You should see something like.
+
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+ annotations:
+   pv.kubernetes.io/provisioned-by: blockstorage.cloudprovider.example
+ creationTimestamp: "2023-02-26T13:14:07Z"
+ finalizers:
+ - kubernetes.io/pv-protection
+ name: pvc-95eb531a-d675-49f6-940b-9bc3fde83eb0
+ resourceVersion: “178817"
+ uid: 6fa824d7-8a06-4e0c-b722-d3f897dcbd65
+spec:
+ accessModes:
+ - ReadWriteOnce
+ capacity:
+   storage: 6Gi
+ claimRef:
+   apiVersion: v1
+   kind: PersistentVolumeClaim
+   name: csi-pvc
+   namespace: default
+   resourceVersion: "178817"
+   uid: 95eb531a-d675-49f6-940b-9bc3fde83eb0
+ csi:
+   driver: blockstorage.cloudprovider.example
+   nodeExpandSecretRef:
+     name: test-secret
+     namespace: default
+   volumeAttributes:
+     storage.kubernetes.io/csiProvisionerIdentity: 1648042783218-8081-blockstorage.cloudprovider.example
+   volumeHandle: e21c7809-aabb-11ec-917a-2e2e254eb4cf
+ nodeAffinity:
+   required:
+     nodeSelectorTerms:
+     - matchExpressions:
+       - key: topology.hostpath.csi/node
+         operator: In
+         values:
+         - racknode01
+ persistentVolumeReclaimPolicy: Delete
+ storageClassName: csi-blockstorage-sc
+ volumeMode: Filesystem
+status:
+ phase: Bound
```

Comment:
Wrap this in <tt>\`\`\`yaml</tt> / <tt>\`\`\`</tt>, and make the first line be `---`.